✨ feat: implement form label generation

de-vasta opened the pull request:
- Add labels to form fields using `humanizeName()` function
- Set input placeholders with capitalized field names
- Update demo link in **README**

de-vasta committed:
✨ feat: implement form label generation

- Add labels to form fields using `humanizeName()` function
- Set input placeholders with capitalized field names
- Update demo link in **README**

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1,3 +1,17 @@
+});
+
+function humanizeName(str) {
+  return str.replace(/([A-Z])/g, ' $1');
```
This function correctly adds a space to separate words in a camelCase string (e.g., `firstName` becomes `first Name`). However, based on the expected result image, the final text for both labels and placeholders should be properly capitalized (e.g., `First name`).

This function should handle the complete transformation. A more robust approach would be to:
1. Add spaces to the camelCase string.
2. Convert the resulting string to lowercase.
3. Capitalize only the first letter.

If this function returns the fully formatted string, you can then use that single result for both the label's text and the input's placeholder, simplifying your main logic.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Great work on this task! Your solution correctly identifies the form fields and dynamically adds the required labels and placeholders. However, I'm requesting a change because there's a small formatting issue with the text capitalization. The generated labels and placeholders (e.g., `first Name`) don't quite match the expected result image (e.g., `First name`). You can fix this by modifying the `humanizeName` function to also handle capitalizing the first letter of the string and ensuring the rest is lowercase. A single adjustment there should resolve the issue for both the labels and placeholders.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

de-vasta committed:
♻️ refactor(main): extract `capitalizeFirstLetter()` utility function

- Replace inline capitalization with reusable function
- Update `humanizeName()` to use `capitalizeFirstLetter()` helper
- Improve code consistency and maintainability

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1,3 +1,23 @@
+    `<label class="field-label" for="${inputEl.id}">${inputName}</label>`,
+  );
+
+  inputEl.placeholder = capitalizeFirstLetter(inputName);
```
Good job on capitalizing the placeholder text. Since the `inputName` variable is already capitalized correctly on line 5, you can set the placeholder directly without calling `capitalizeFirstLetter` again.

For example:
`inputEl.placeholder = inputName;`

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Excellent work on this pull request! I am approving your solution as you've successfully addressed the feedback from the previous review and met all the task requirements. The capitalization for labels and placeholders now functions correctly. As a minor suggestion for future improvement, you can make the code in `src/scripts/main.js` slightly more efficient by removing the redundant `capitalizeFirstLetter` call when setting the placeholder, since `inputName` is already capitalized. You can simply use `inputEl.placeholder = inputName;`. This is just a small tip, and overall you've done a great job!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
